raster-foundry/raster-foundry
# Add platform and org group roles when a user is created

## Overview

Store default platform / org ids in env variables
When creating a user, check for platform / organization ids defined on the id_token


### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] [Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~
~- [ ] Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~

### TODO
- [x] make it use the default_organization_id field on the platform where it makes sense, once the pr has been merged

## Testing Instructions
* Using psql, create up a new platform and org
* In you .env file, add the following:
```
AUTH0_SYSTEM_USER=auth0|5a0f360178914308349ac641
AUTH0_DEFAULT_PLATFORM_ID={new platform id}
AUTH0_DEFAULT_ORGANIZATION_ID={new org id}
```
* rsync / run the api server
* Use the overrides file to force the frontend to use the dev-alt tenant (see https://github.com/raster-foundry/raster-foundry/pull/3332 for instructions)
* Verify that the alt tenant has a rule which adds a default platform / org
![image](https://user-images.githubusercontent.com/4392704/40245142-bef90868-5a92-11e8-9507-aeb6f9cf79c6.png)

* Create a new user on the alt tenant
* Log in with the new user
* verify that roles for the correct org and platform are created, and that they are created by the system user
Closes #3314 
Depends https://github.com/raster-foundry/raster-foundry/pull/3394
